js/mushino.js

```diff
@@ -0,0 +1,672 @@
+        };
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const stopPrice = this.safeFloat (params, 'stopPrice');
+        const uppercaseType = type.toUpperCase ();
+        let priceIsRequired = false;
+        let triggerPriceIsRequired = false;
+        const method = 'privatePostAuthOrder';
+        const order = {
+            'pair': market['id'],
+            'qty': this.amountToPrecision (symbol, amount),
+            'side': side.toUpperCase (),
+        };
+        if (uppercaseType === 'LIMIT') {
+            order['price'] = price;
+            priceIsRequired = true;
+        } else if (uppercaseType === 'STOP_LOSS') {
+            order['trigger_price'] = stopPrice;
+            triggerPriceIsRequired = true;
+        } else if (uppercaseType === 'STOP_LOSS_LIMIT') {
+            order['trigger_price'] = stopPrice;
+            order['price'] = price;
+            triggerPriceIsRequired = true;
+            priceIsRequired = true;
+        } else if (uppercaseType === 'TAKE_PROFIT') {
+            order['trigger_price'] = stopPrice;
+            triggerPriceIsRequired = true;
+        } else if (uppercaseType === 'TAKE_PROFIT_LIMIT') {
+            order['trigger_price'] = stopPrice;
+            order['price'] = price;
+            triggerPriceIsRequired = true;
+            priceIsRequired = true;
+        }
+        if (priceIsRequired && price === undefined) {
+            throw new InvalidOrder ('createOrder method requires a price argument for a ' + type + ' order');
+        }
+        if (triggerPriceIsRequired && stopPrice === undefined) {
+            throw new InvalidOrder ('createOrder method requires a trigger_price as an extra param for a ' + type + ' order');
+        }
+        const response = await this[method] (this.extend (order, params));
+        return this.parseOrder (response['result']['order'], market);
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        const response = await this.privateGetBalances (this.extend (request, params));
+        const balances = response['result'];
+        const result = { 'info': balances };
+        const currencies = Object.keys (balances);
+        for (let i = 0; i < currencies.length; i++) {
+            const currencyId = currencies[i];
+            const balance = balances[currencyId];
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            const free_cash = this.safeFloat (balance, 'free');
+            const locked_cash = this.safeFloat (balance, 'locked');
+            const withdrawn_cash = this.safeFloat (balance, 'withdrawn');
+            const free_margin = this.safeFloat (balance, 'margin');
+            const free_total = free_cash + (free_margin || 0);
```

Comment:
https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings